### PR TITLE
allow user setting to control extended timeout, reduce default to 55 …

### PR DESCRIPTION
**Pull request by coury-clark**
Allow users to override the extended timeout, as well as defaulting to 55 instead of 60.

## Test plan

Testing a long running query with a low value set for `insights.aggregations.extendedTimeout` then retry with a longer timeout.

CleanShot 2022-09-08 at 13 58 24

### CODE REVIEW

File: enterprise/internal/insights/resolvers/aggregates_resolvers.go [hunk 1]
```diff
@@ -153,6 +153,18 @@ func (r *searchAggregateResolver) Aggregations(ctx context.Context, args graphql
 	}}, nil
 }
 
+func getExtendedTimeout(ctx context.Context, db database.DB) int {
+	settings, err := graphqlbackend.DecodedViewerFinalSettings(ctx, db)
+	if err != nil || settings == nil {
+		return extendedSearchTimeLimitSecondsDefault
+	}
+	val := settings.InsightsAggregationsExtendedTimeout
+	if val > 0 {
```
Comment (chwarwick) [inline enterprise/internal/insights/resolvers/aggregates_resolvers.go:162]: maybe we should put a cap on this. Maybe not to exceed 120 seconds?

**Review by chwarwick — APPROVED** [review]
Think a max is wise to avoid any accidentally long settings.

File: enterprise/internal/insights/resolvers/aggregates_resolvers.go (hunk 1 above)
Comment (coury-clark) [reply enterprise/internal/insights/resolvers/aggregates_resolvers.go:173]: The max value is now the same as the max search timeout. https://sourcegraph.slack.com/archives/C02KE7E9GMR/p1662680231417329?thread_ts=1662677310.566389&cid=C02KE7E9GMR

File: enterprise/internal/insights/resolvers/aggregates_resolvers.go [hunk 2]
```diff
@@ -153,6 +155,27 @@ func (r *searchAggregateResolver) Aggregations(ctx context.Context, args graphql
...
+	searchLimit := limits.SearchLimits(conf.Get()).MaxTimeoutSeconds
+
+	min := func(x, y int) int {
+		if x < y {
+			return x
+		}
+		return y
+	}
+
+	settings, err := graphqlbackend.DecodedViewerFinalSettings(ctx, db)
```
[the remark this answers is not in the document]
Comment (coury-clark) [reply enterprise/internal/insights/resolvers/aggregates_resolvers.go:168]: the idea is for this to be in user settings, so an individual can override it as necessary

File: enterprise/internal/insights/resolvers/aggregates_resolvers_test.go [hunk 3]
```diff
@@ -481,3 +481,7 @@ func Test_buildDrilldownQuery(t *testing.T) {
 		})
 	}
 }
+
+func TestThing(t *testing.T) {
+
+}
```
[the remark this answers is not in the document]
Comment (coury-clark) [reply enterprise/internal/insights/resolvers/aggregates_resolvers_test.go:487]: whoops! 😄

**Pull request merged**